libevmasm/SimplificationRules.cpp

```diff
@@ -126,21 +131,31 @@ Rules::Rules()
 
 		// invariants involving known constants
 		{{Instruction::ADD, {X, 0}}, [=]{ return X; }},
+		{{Instruction::ADD, {0, X}}, [=]{ return X; }},
```

Comment:
This rule should already be present. All operators that are known to be commutative are applied a second time to swapped arguments.